src/HRQLS/Models/ElasticSearchServiceProvider.php

```diff
@@ -82,12 +84,188 @@ public function boot(Application $app)
     /**
      * Search functionality exposed for use through the service provider.
      *
-     * @param array $req The request to be sent to the ElasticSearch client.
+     * @param array $indices The list of indices to include in the query. Defaults to searching all indices.
+     * @param array $types   The list of types to include in the query. Defaults to searching all types.
+     * @param array $query   The query to run against the specified indexes and types.
      *
-     * @return array
+     * @return array Like [
+     *    'took' => The amount of time the query took to execute,
+     *    'timed_out' => [
+     *      '_shards' => [
+     *        'total' => (Integer), Total number of shards queried,
+     *        'successful' => (integer), Number of successfully queried shards.
+     *        'failed' => (integer), Number of shards that could not be queried.
+     *    ],
+     *    'hits' => [
+     *      'total' => (integer), Number of documents returned by the query.
+     *      'max_score' => (double), The highest score assigned during execution.
+     *      'hits' => [
+     *      ],
+     *    ],
+     *  ];
      */
-    public function search(array $req)
+    public function search(array $indices, array $types, array $query = [])
     {
+        if (empty($indices)) {
```

Comment:
I was going to yell about not setting a default value in the parameter list... but I think this actually looks cleaner.